src/components/BottomBar/Origin/OriginPicker.jsx

```diff
@@ -2,98 +2,225 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 
 import SmallLabel from '../../common/SmallLabel/SmallLabel';
-import Icon from '../../common/Icon/Icon';
 import LoadingString from '../../common/LoadingString/LoadingString';
 import Picker from '../Picker';
 import Popover from '../../common/Popover/Popover';
+import Button from '../../common/Input/Button/Button';
 import FilterList from '../../common/FilterList/FilterList';
-import DataManager from '../../../api/DataManager';
-import { NavigationAnchorKey, NavigationAimKey, RetargetAnchorKey, RetargetAimKey } from '../../../api/keys';
+import {
+  NavigationAnchorKey,
+  NavigationAimKey,
+  RetargetAnchorKey,
+  RetargetAimKey,
+  ScenePrefixKey
+} from '../../../api/keys';
 import FocusEntry from './FocusEntry';
 
-import Earth from './images/earth.png';
+import {
+  setNavigationAction, 
+  setOriginPickerShowFavorites,
+  setPopoverVisibility
+} from '../../../api/Actions';
+
 import styles from './OriginPicker.scss';
 
-const icons = {
-  Earth,
-};
+import SvgIcon from '../../common/SvgIcon/SvgIcon';
+import MaterialIcon from '../../common/MaterialIcon/MaterialIcon';
+import Anchor from 'svg-react-loader?name=Anchor!../../../icons/anchor.svg';
+import Aim from 'svg-react-loader?name=Aim!../../../icons/aim.svg';
+import Focus from 'svg-react-loader?name=Focus!../../../icons/focus.svg';
+
+import propertyDispatcher from '../../../api/propertyDispatcher';
+import subStateToProps from '../../../utils/subStateToProps';
+
 
 // tag that each focusable node must have
-const REQUIRED_TAG = 'planet_solarSystem';
+const REQUIRED_TAG = 'GUI.Interesting';
+
+const NavigationActions = {
+  Focus: 'Focus',
+  Anchor: 'Anchor',
+  Aim: 'Aim'
+};
 
 class OriginPicker extends Component {
   constructor(props) {
     super(props);
-
-    this.state = {
-      origin: 'Mercury',
-      hasOrigin: false,
-      sceneGraphNodes: [],
-      showPopover: false,
-    };
-
-    this.updateOrigin = this.updateOrigin.bind(this);
     this.togglePopover = this.togglePopover.bind(this);
+    this.onSelect = this.onSelect.bind(this);
   }
 
   componentDidMount() {
-    DataManager.subscribe(NavigationAnchorKey, this.updateAnchor);
-    DataManager.subscribe(NavigationAimKey, this.updateAim);
+    this.props.anchorDispatcher.subscribe();
+    this.props.aimDispatcher.subscribe();
   }
 
   componentWillUnmount() {
-    DataManager.unsubscribe(NavigationAnchorKey, this.updateAnchor);
-    DataManager.unsubscribe(NavigationAimKey, this.updateAim);
+    this.props.anchorDispatcher.unsubscribe();
+    this.props.aimDispatcher.unsubscribe();
   }
 
-  get icon() {
-    const icon = icons[this.state.origin];
-    if (icon) {
-      return (
-        <img src={icon} className={styles.iconImage} alt={this.state.origin} />
-      );
-    }
```

Comment:
storyTouch version might need this?

Reply:
This code is not used in storyTouch - this is a regular webgui component